### ISSUE TITLE
Add support for fasthttp

### ISSUE THREAD

**Issue by valyala** [googollee/go-socket.io#127]
It would be great if [fasthttp](https://github.com/valyala/fasthttp) support is added to `go-socket.io` additionally to [net/http](https://golang.org/pkg/net/http/). See how [this websocket library](https://github.com/leavengood/websocket) is implemented on top of `fasthttp` via [RequestCtx.Hijack](https://godoc.org/github.com/valyala/fasthttp#RequestCtx.Hijack).

**Comment by CyberLight** [comment]
@valyala i think what it's not a good idea for supporting fasthttp because it's third party library and may be not supported by maintaners (by you for example). But net/http supported by golang team and it's can be modified for speed and perfomance.

This [Websocket library](https://github.com/leavengood/websocket) used in production?
Where is benchmarks, metrics of real usage?

**Comment by valyala** [comment]
> This Websocket library used in production?
> Where is benchmarks, metrics of real usage?

I don't know. @leavengood , the author of the package, should know better.

**Comment by leavengood** [comment]
My websocket library is just a fork of the [gorilla websocket library](https://github.com/gorilla/websocket), as clearly indicated in the README. I just added support for fasthttp, but the gorilla project did not want to add that for similar reasons as you, so I just decided to keep it in my own fork.

This was for a prototype work project which has currently been put on hold, which is why I have not been keeping my fork up-to-date with the gorilla version.

Adding support for fasthttp was very easy, and just involved writing my own HTTP upgrade handler. All the other websocket handling is still from the core code of gorilla websockets.
